test/unistd/write_broken_link.c

```diff
@@ -0,0 +1,49 @@
+#include <stdlib.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+
+char* join_path(const char* path1, const char* path2) {
+    int len1 = strlen(path1);
+    int len2 = strlen(path2);
+    char* result = malloc(len1 + len2 + 2);
+    memcpy(result, path1, len1);
+    result[len1] = '/';
+    memcpy(result + len1 + 1, path2, len2);
+    return result;
+}
+
+int main() {
+    char template[] = "/tmp/tmpdir.XXXXXX";
+    char *tmpdir = mkdtemp(template);
```

Comment:
Let not use `mkdtemp` here, but something local like `./working`.  That also avoids malloc and free as p1 and p2 can then simply be string constants.

And we can simply delete `join_path`